profiler/options.go

```diff
@@ -557,3 +559,18 @@ func (e *executionTraceConfig) Refresh() {
 	// if it's udpated to be invalid
 	e.warned = false
 }
+
+// WithCustomProfilerLabels specifies [profiler label] keys which should be
+// available as attributes for filtering frames for CPU and goroutine profile
+// flame graphs in the Datadog profiler UI.
+//
+// The profiler is limited to 20 label keys to show in the UI. Any label keys
+// after the first 20 will be ignored (but labels with ignored keys will still
+// be available in the raw profile data)
+//
+// [profiler label]: https://rakyll.org/profiler-labels/
+func WithCustomProfilerLabels(keys ...string) Option {
```

Comment:
I think we should rename this func to `WithCustomProfilerLabelKeys`. Otherwise the API and the linked article make it easy to get confused and assume that this API works the same ways as `pprof.Labels()` which expects key value pairs.